store/tikv/lock_resolver.go

```diff
@@ -578,10 +595,235 @@ func (lr *LockResolver) getTxnStatus(bo *Backoffer, txnID uint64, primary []byte
 			status.commitTS = cmdResp.CommitVersion
 			lr.saveResolved(txnID, status)
 		}
+
 		return status, nil
 	}
 }
 
+// asyncResolveData is data contributed by multiple goroutines when resolving locks using the async commit protocol. All
+// data should be protected by the mutex field.
+type asyncResolveData struct {
```

Comment:
We could add some functions for `asyncResolveData` to mark the current transaction status like `SetCommit` or `SetRollback` if the status could be decided by the protocol. And some comments could be added to explain why it could be decided under certain conditions. Also, we need some logs to trace the decision so that it's easy to trace the transaction behaviors doing debug.

Reply:
> We could add some functions for asyncResolveData to mark the current transaction status like SetCommit or SetRollback if the status could be decided by the protocol

It is not this simple, unfortunately, essentially TiKV decides and we just store the commit ts which encodes it.

> And some comments could be added to explain why it could be decided under certain conditions.

I'll try and write some...

> Also, we need some logs to trace the decision so that it's easy to trace the transaction behaviors doing debug.

:+1: